src/hotspot/share/gc/g1/g1Allocator.hpp

```diff
@@ -145,6 +145,22 @@ class G1Allocator : public CHeapObj<mtGC> {
                                    uint node_index);
 };
 
+// Specialized PLAB for old generation promotions. For old regions the
+// BOT needs to be updated and the relevant data to do this efficiently
+// is stored in the PLAB.
+class G1BotUpdatingPLAB : public PLAB {
+  // An object spanning this threshold will cause a BOT update.
+  HeapWord* _next_bot_threshold;
+  // The region in which the PLAB resides.
+  HeapRegion* _region;
+public:
+  G1BotUpdatingPLAB(size_t word_sz) : PLAB(word_sz) { }
+  // Sets the new PLAB buffer as well as updates the threshold and region.
+  virtual void set_buf(HeapWord* buf, size_t word_sz);
```

Comment:
Is `override` better here?

Reply:
I can change this to `override` if you prefer. I see that we have some uses of it in G1 already.

Reply:
Yes please, since it makes the intention explicit and enforces it in compile-time.